server.js

```diff
@@ -1,39 +1,172 @@
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// Defining models 
+const bookSchema = new mongoose.Schema({
+  bookId: Number,
+  title: {
+    type: String,
+    lowercase: true
+  },
+  authors: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Author',
+    lowercase: true
+  },
+  average_rating: Number,
+  isbn: String,
+  language_code: {
+    type: String,
+    lowercase: true
+  },
+  num_pages: Number,
+  ratings_count: Number
+})
+
+const Book = mongoose.model('Book', bookSchema)
+
+const Author = mongoose.model('Author', {
+  authors: String
+})
+
+// The seedDB is going to re-run everytime we start the server, if we don't want that
+// to happen and want to be able to choose when that should happen, we can wrap it
+// in an environment variable
+if (process.env.RESET_DATABASE) {
+  const seedDB = async () => {
+    await Book.deleteMany()
+    await Author.deleteMany()
+
+    const authorsArray = []
+
+    booksData.forEach(async (item) => {
+      const author = new Author(item)
+      authorsArray.push(author)
+      await author.save()
+    })
+
+    booksData.forEach(async (item) => {
+      const newBook = new Book({
+        ...item,
+        authors: authorsArray.find(singleAuthor => singleAuthor.authors === item.authors)
+      })
+      await newBook.save()
+    })
+  }
+  seedDB()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
+
+// Middleware - since it is mounted before the route, this middlewear will execute first. 
+// It checks if the database is in a good state or not. If so, invoke next() and continue on to show our data, if not show error message.
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
 
-// Start defining your routes here
+// Route to APIs first page listing all possible endpoints
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
 })
 
-// Start the server
-app.listen(port, () => {
-  // eslint-disable-next-line
-  console.log(`Server running on http://localhost:${port}`)
+// Endpoint to get all books 
+app.get('/books', async (req, res) => {
+  const books = await Book.find().populate('authors')
+  res.json({ length: books.length, data: books })
 })
+
+// Endpoint to get single book by id path param
+app.get('/books/book/:bookId', async (req, res) => {
```

Comment:
`/books/:bookId` - id is a primary key so you should add it immediately after set name in your endpoint